x-pack/test/functional/services/ml/job_table.ts

```diff
@@ -50,6 +50,28 @@ export function MachineLearningJobTableProvider(
   const retry = getService('retry');
 
   return new (class MlJobTable {
+    public async clickJobRowCalendar(jobId: string, calendarId: string) {
+      await this.withDetailsOpen(jobId, async function clickJobRowCalendar() {
+        const calendarSelector = `${jobId}-${calendarId}`;
+        await testSubjects.existOrFail(calendarSelector, {
+          timeout: 3_000,
+        });
+        await testSubjects.click(calendarSelector, 3_000);
+      });
+    }
```

Comment:
@wayneseymour `withDetailsOpen` first calls `ensureDetailsOpen` then runs the code block and then calls `ensureDetailsClosed`. The last one obviously can't work if the code block navigated to a different page.

For the implementation details: `ensureDetailsClosed` checks 
```
if (await testSubjects.exists(this.detailsSelector(jobId))) {
```
so in case the block code navigated to a different page, this method doesn't do anything (as the row details selector doesn't exist on that other page). But if that implementation was ever replaced by something that expects the jobs table and the job row to exist, then things would fail in a weird way. `withDetailsOpen` should really only be used for operations on that page. For everything else, `ensureDetailsOpen` should work fine.

Does that answer your question?

Reply:
What do you think about [this](https://github.com/elastic/kibana/pull/178553/commits/fd23d1590d8cbb6378885434bb37cb09e00c3fc1) boss?